automata/fa/dfa.py

```diff
@@ -438,17 +539,21 @@ def _minify(
         new_transitions = {
             name: {
                 letter: back_map[transitions[next(iter(eq))][letter]]
-                for letter in input_symbols
+                for letter in transitions[next(iter(eq))].keys()
             }
             for name, eq in eq_class_name_pairs
         }
 
+        allow_partial = any(
+            len(lookup) != len(input_symbols) for lookup in transitions.values()
```

Comment:
@EduardoGoulart1 @eliotwrobson I think a DFA can also be considered partial if the top-level transitions dict doesn't include all of the states as its keys (i.e. any states not listed are trap states). For example:

```python
# A DFA whose language is just {"a"}
dfa = DFA(
    states={"q0", "q1"},
    input_symbols={"a", "b"},
    transitions={
        # This case is already covered in the above check, since the transition to "b" is left out
        "q0": {"a": "q1"}
        # Definition for "q1" is omitted because it has no transitions anyway;
        # this case is not considered in the above check to determine a DFA"s partiality
    },
    initial_state="q0",
    final_states={"q1"},
    allow_partial=True,
)
```

Please correct me if I misunderstand the code here, but it just caught my eye.

Reply:
@caleb531 this functionality was removed here, take a look at the commit message: https://github.com/EduardoGoulart1/automata/commit/43f377c4b7aa6381c286679a0379776857e2459c

I don't think it's worth adding support for this, since it would lead to some incompatibilities with existing code, and we'd have to go back and test some extra things. This only leads to minor memory savings in very limited circumstances anyway.

Reply:
@eliotwrobson Okay, fair enough. The current API documentation doesn't say anything against your recommended behavior anyway

> `allow_partial`: by default, each DFA state must have a transition to
every input symbol; if `allow_partial` is `True`, you can disable this
characteristic (such that any DFA state can have fewer transitions than input
symbols)